mesonbuild/modules/windows.py

```diff
@@ -75,14 +75,26 @@ def _find_resource_compiler(self, state: 'ModuleState') -> T.Tuple[ExternalProgr
         rescomp = ExternalProgram.from_bin_list(state.environment, for_machine, 'windres')
 
         if not rescomp or not rescomp.found():
+            def search_programs(names: T.List[str]) -> T.Optional[ExternalProgram]:
+                for name in names:
+                    program = ExternalProgram(name, silent=True)
+                    if program.found():
+                        return program
+                return None
+
+            def find_rc_like_compiler() -> T.Optional[ExternalProgram]:
+                return search_programs(['rc', 'llvm-rc'])
+
+            def find_windres_like_compiler() -> T.Optional[ExternalProgram]:
+                return search_programs(['windres', 'llvm-windres'])
+
             comp = self.detect_compiler(state.environment.coredata.compilers[for_machine])
             if comp.id in {'msvc', 'clang-cl', 'intel-cl'} or (comp.linker and comp.linker.id in {'link', 'lld-link'}):
-                # Microsoft compilers uses rc irrespective of the frontend
-                rescomp = ExternalProgram('rc', silent=True)
+                rescomp = find_rc_like_compiler() or find_windres_like_compiler()
```

Comment:
The commit message doesn't rationalize why for MSVC/clang-cl we now search for windres